pipeline-model-definition/src/main/resources/org/jenkinsci/plugins/pipeline/modeldefinition/agent/impl/DockerPipelineFromDockerfileScript.groovy

```diff
@@ -48,14 +49,30 @@ public class DockerPipelineFromDockerfileScript extends DeclarativeAgentScript {
             if (!Utils.withinAStage()) {
                 script.stage(SyntheticStageNames.agentSetup()) {
                     Utils.markSyntheticStage(SyntheticStageNames.agentSetup(), Utils.getSyntheticStageMetadata().pre)
-                    buildImage().call()
+                    try {
+                        buildImage().call()
+                    } catch (Exception e) {
+                        script.getProperty("currentBuild").result = Result.FAILURE
```

Comment:
Shouldn't there be some information printed to the build log explaining that building the image failed?